bit_manipulation/missing_number.py

```diff
@@ -0,0 +1,21 @@
+def find_missing_number(nums):
+    """
+    Finds the missing number in a list of consecutive integers.
+
+    Args:
+        nums (List[int]): A list of integers.
+
+    Returns:
+        int: The missing number.
```

Comment:
[Mypy](mypy.readthedocs.io) can test data types in the function signature, but not the comments.

Do not repeat the datatypes in both places because readers will be confused if one is changed and the other is not.

```suggestion
def find_missing_number(nums: list[int]) -> int:
    """
    Finds the missing number in a list of consecutive integers.

    Args:
        nums: A list of integers.

    Returns:
        The missing number.
```